src/main/java/frc/robot/subsystems/Arm.java

```diff
@@ -59,7 +55,28 @@ public class CoralArm extends SubsystemBase {
   private double currentAngleTime = 0;
 
   /** Creates a new Arm. */
-  public CoralArm() {
+  public Arm(ArmParameters PARAMETERS) {
+
+    MASS = PARAMETERS.getMass();
+    GEAR_RATIO = PARAMETERS.getGearRatio();
+    MOTOR = PARAMETERS.getMotorParameters();
+
+    EFFICIENCY = PARAMETERS.getEfficiency();
+    RADIANS_PER_REVOLUTION = PARAMETERS.getRadiansPerRevolution();
+    MAX_ANGULAR_SPEED = EFFICIENCY * MOTOR.getFreeSpeedRPM() * RADIANS_PER_REVOLUTION / 60.0;
+    ARM_LENGTH = PARAMETERS.getArmLength();
+    MAX_ANGULAR_ACCELERATION =
+        EFFICIENCY * ((2 * MOTOR.getStallTorque() * GEAR_RATIO) / (MASS * ARM_LENGTH));
+    CONSTRAINTS =
+        new TrapezoidProfile.Constraints(MAX_ANGULAR_SPEED * 0.3, MAX_ANGULAR_ACCELERATION * 0.5);
+    KS = PARAMETERS.getkS();
```

Comment:
The `feedForward` and `controller` member variables are initialized in their declaration but depend on values initialized in the constructor. Initialization is done before the constructor runs, so the `Kx` and other values are indeterminate. (Does this PR build?)

Since we only use the `Kx` values to initialize the `ArmFeedforward` and the values are determined solely by the arm parameters, we could just add a method to create the `ArmFeedforward` in the `ArmParameters` enum.

```java
public ArmFeedforward getArmFeedforward() {
  ...
}

The PID constants may need to change between the coral and algae arm, so we should add a method to create the `ProfiledPIDController` to the `ArmParameters` enum as well.
```